www/source/blog/2017-10-05-Introducing-Composites.html.md

```diff
@@ -0,0 +1,274 @@
+---
+title: Introducing Composites
+date: 2017-10-05
+author: Christopher Maier
+tags: blog
+category: supervisor, packaging
+classes: body-article
+---
+
+We're excited to talk about a new feature in Habitat that we feel will
+unlock some more interesting use cases that community members have
+been wanting for a while. In this post, we'll talk about _composite
+packages_ and how they might help you.
+
+## What is a Composite Package?
+
+As the name might suggest, a _composite package_ is a special kind of Habitat package that includes other Habitat packages, specifically _services_. A service is something you would actually run in a Habitat Supervisor: something like [core/redis](https://bldr.habitat.sh/#/pkgs/core/redis/latest), but not a library like, say, [core/linux-headers](https://bldr.habitat.sh/#/pkgs/core/linux-headers/latest). Composite packages allow you to group together services that should be run together on the same Supervisor, enabling you to take better advantage of [modern deployment patterns](http://blog.kubernetes.io/2015/06/the-distributed-system-toolkit-patterns.html) with your Habitat services.
+
+## Why?
+
+In the past, the intrepid Habitat user community has been finding all sorts of creative ways to use Habitat in a variety of ways. Since the sidecar pattern is becoming increasingly prevalent, particularly with microservice architectures and containerized deployments, users have tried to implement it in Habitat, but with limited success. Often, we would see users trying to start multiple services from the `init` or `run` hook of another coordinating service. Approaches like this may work initially, but won't work in the long run, because the Habitat Supervisor won't be properly supervising additional services that are spawned this way.
```

Comment:
I might link off to the sidecar pattern or an example of it here

Reply:
Might be worth calling out here that any calls to `hab` is undefined behaviour in a hook